flink-core/src/main/java/org/apache/flink/util/ExceptionUtils.java

```diff
@@ -109,6 +131,26 @@ public static boolean isJvmFatalOrOutOfMemoryError(Throwable t) {
 		return isJvmFatalError(t) || t instanceof OutOfMemoryError;
 	}
 
+	/**
+	 * Checks whether the given exception indicates a JVM metaspace out-of-memory error.
+	 *
+	 * @param t The exception to check.
+	 * @return True, if the exception is the metaspace {@link OutOfMemoryError}, false otherwise.
+	 */
+	public static boolean isMetaspaceOutOfMemoryError(Throwable t) {
+		return t instanceof OutOfMemoryError && t.getMessage().startsWith("Metaspace");
+	}
```

Comment:
Good question.
I have not found any other ways to do it using some Java API.
The message is documented though, e.g. in Oracle Java [8](https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/memleaks002.html) and [11](https://docs.oracle.com/en/java/javase/11/troubleshoot/troubleshoot-memory-leaks.html#GUID-19F6D28E-75A1-4480-9879-D0932B2F305B).

Reply:
What about openjdk? Could we add tests for these assumption so that we are sure if you build and run the tests on your JVM, then these assumptions hold?

Reply:
True, this should help

Reply:
I added the Metaspace assumption test. It is a bit tricky to make sure and load all the classes to measure the Metaspace usage so that OOM error is thrown in expected class loading. It was stable locally. If this ever leads to an instability or you also have more doubts, I can remove the commit.